### DISCUSSION TITLE
->label()-fuction on modal-forms

### DISCUSSION THREAD

**Discussion by Husky110** [filamentphp/filament#7015] [Ideas]
Hi,  
Right now it is only possible to modify a modal-form-header by setting the `getModelLabel` and `getPluralModelLabel`-functions. It would be great to have a `->label()`-function like in most components and items to customize the header of a modal-form inside a resource.  
And please add that to v2 aswell. :)

**Comment by danharrin** [comment]
`->modalHeading()`?

**Comment by Husky110** [comment]
@danharrin - Gives me this:
![image](https://github.com/filamentphp/filament/assets/13180625/d2ea5c4b-a451-47a4-ba1a-3782dde62fde)

Tried it via
![image](https://github.com/filamentphp/filament/assets/13180625/307e9b2d-0b86-4884-ab93-00d06a56d273)

I'm talking about "inside the resource" here. :)

**Reply by danharrin** [reply]
`EditAction::make()->modalHeading()` etc

**Reply by Husky110** [reply]
But when I already have a form-function, why not make it available there?

**Reply by danharrin** [reply]
Because its nothing to do with the form, its to do with the action modal. It's not appropriate there. And you'd still need a way to customize it per-action.

**Reply by Husky110** [reply]
Okay - I get what you are saying. But on the other hand you're making the form-function available to basically make the edit-action beeing setup easily and in this I think it is appropriate, cause I don't see the point in the form()-function at all then, when the "good" way would be to setup an edit-action.

**Reply by danharrin** [reply]
The resource is a translation layer between whatever UI you choose - pages or modals. It should not be opinionated towards one.

**Reply by Husky110** [reply]
So - just so I get you right: The form-function is kinda like "hey - you can use it, but you don't have to"? Cause if so, it's kinda weird that it's auto-generated.  
When I see something beeing auto-generated I assume that it's like "you should use what has been generated for you". :)

**Reply by danharrin** [reply]
`form()` is used to configure the form builder. `table()` is used to configure the table builder. Actions not part of the form builder.

**Reply by Husky110** [reply]
Ah okay - now (I think) I get what you are saying! So the flow is to build the form inside the form-function and then create the actions which then use the previously defined forms, where you then have the modalHeading-function. Makes sense.

**Comment by Husky110** [comment]
To clarify: I'm not talking about beeing "not able to change the header at all". I'm talking about making it more intuitive to use. :)

**Comment by Husky110** [comment]
I got the workflow wrong - thanks again Dan for the clarification. :)
